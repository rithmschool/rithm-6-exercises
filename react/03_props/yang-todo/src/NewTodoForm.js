import React, { Component } from 'react';
import Todo from './Todo.js'

export default class NewTodoForm extends Component {

    render() {
        return(
            <form onSubmit={this.props.addTodo}>
                <label> Todo List </label>
                <input type="text" name="task"/>
                <br/>
                <input type="submit" value="Submit"/>
            </form>
        )
    }
}