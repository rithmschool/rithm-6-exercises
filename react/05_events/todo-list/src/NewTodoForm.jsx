import React, { Component } from "react";

class TodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = { val: "" };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.handleAdd(this.state.val);
    this.setState({ val: "" });
  }
  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input name="val" onChange={this.handleChange} value={this.state.val} />
        <input type="submit" />
      </form>
    );
  }
}

export default TodoForm;
