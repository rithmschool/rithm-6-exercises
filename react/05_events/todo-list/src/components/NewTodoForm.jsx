import React, { Component } from "react";

class NewTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = { val: "" };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.addTodo(this.state.val);
    this.setState({ val: "" });
    this.props.history.push("/");
  }
  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input name="val" onChange={this.handleChange} value={this.state.val} />
        <input type="submit" />
      </form>
    );
  }
}

export default NewTodoForm;
