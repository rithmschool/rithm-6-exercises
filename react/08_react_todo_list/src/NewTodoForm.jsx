import React, { Component } from 'react';

class NewTodoForm extends Component {
    handleSubmit(event) {
        event.preventDefault();
        this.props.addTodo({
            title: event.target.title.value,
            description: event.target.description.value,
        });
        event.target.reset();
    }
    
    render() {
        return (
            <form className={this.props.isOpen ? "mx-auto my-3 text-left todo-form" : "mx-auto my-3 text-left todo-form hide"}  onSubmit={this.handleSubmit.bind(this)}>
                <div className="form-group">
                    <label className="mb-0">Title</label>
                    <input type="text" className="form-control" id="title" name="title" placeholder="Enter title of todo" required />
                    <small id="titleHelp" className="form-text text-muted">You can enter any title for the todo</small>
                </div>
                <div className="form-group">
                    <label className="mb-0">Description</label>
                    <input type="text" className="form-control" id="description" name="description" placeholder="Enter description of todo" />
                    <small id="descriptionHelp" className="form-text text-muted">You can enter any description for the todo</small>
                </div>
                <button type="submit" className="btn btn-secondary btn-block">Add new todo</button>
            </form>
        )
    }
}

export default NewTodoForm;