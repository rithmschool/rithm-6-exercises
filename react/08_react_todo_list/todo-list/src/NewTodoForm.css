.todo__area {
  height: 25px;
  width: 500px;
  border-radius: 5px;
  margin: 10px;
}
