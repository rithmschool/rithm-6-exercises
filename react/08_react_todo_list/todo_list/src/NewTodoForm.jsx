import React, { Component } from 'react';

export default class NewTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = { title: '', description: '', isComplete: false };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  handleSubmit(e) {
    e.preventDefault();
    this.props.handleAdd(this.state);
    this.setState({ title: '', description: '', isComplete: false });
    e.target.reset();
  }
  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input placeholder="Enter New Todo" onChange={this.handleChange} name="title" type="text" />
        <input
          placeholder="Enter Description"
          onChange={this.handleChange}
          name="description"
          type="text"
        />
        <input type="submit" />
      </form>
    );
  }
}
