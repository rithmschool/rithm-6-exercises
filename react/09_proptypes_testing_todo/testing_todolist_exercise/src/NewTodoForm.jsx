import React, { Component } from "react";
import { PropTypes } from "prop-types";

export default class NewTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: "",
      description: ""
    }
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.handleAdd({ ...this.state });
    this.setState({ title: "", description: "" });
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="title">Title</label>
        <input name="title" value={this.state.title} onChange={this.handleChange} />
        <label htmlFor="description">Description</label>
        <input name="description" value={this.state.description} onChange={this.handleChange} />
        <button>Submit</button>
      </form>
    );
  }

}

NewTodoForm.propTypes = {
  handleAdd: PropTypes.func
};
