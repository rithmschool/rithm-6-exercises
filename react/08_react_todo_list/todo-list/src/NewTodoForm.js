import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./NewTodoForm.css";

class NewTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: ""
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  handleSubmit(e) {
    e.preventDefault();
    this.props.handleAdd({ ...this.state });
    this.setState({ title: "" });
    this.props.history.push("/");
  }
  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input
            className="todo__area"
            name="title"
            value={this.state.title}
            onChange={this.handleChange}
            type="text"
          />
          <input type="submit" />
        </form>
        <button>
          <Link to="/">Check Out All Those Todos</Link>
        </button>
      </div>
    );
  }
}

export default NewTodoForm;
