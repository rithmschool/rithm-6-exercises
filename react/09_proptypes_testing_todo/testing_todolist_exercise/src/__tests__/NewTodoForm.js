import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewTodoForm from '../NewTodoForm';

describe('<NewTodoForm />', () => {
  let wrapper;

  it('render', () => {
    wrapper = shallow(<NewTodoForm />);
  });

  it('matches snapshot', () => {
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  })

})

