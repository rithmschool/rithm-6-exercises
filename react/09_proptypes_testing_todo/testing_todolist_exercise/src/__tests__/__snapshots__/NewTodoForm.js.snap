// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewTodoForm /> matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="title"
  >
    Title
  </label>
  <input
    name="title"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="description"
  >
    Description
  </label>
  <input
    name="description"
    onChange={[Function]}
    value=""
  />
  <button>
    Submit
  </button>
</form>
`;
